pkg/stanza/fileconsumer/file_threadpool.go

```diff
@@ -0,0 +1,206 @@
+package fileconsumer // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer"
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"go.uber.org/zap"
+)
+
+type ReaderWrapper struct {
+	reader *Reader
+	fp     *Fingerprint
+}
+
+// poll checks all the watched paths for new entries
+func (m *Manager) pollConcurrent(ctx context.Context) {
+	// Increment the generation on all known readers
+	// This is done here because the next generation is about to start
+
+	m.knownFilesLock.Lock()
+	for i := 0; i < len(m.knownFiles); i++ {
+		m.knownFiles[i].generation++
+	}
+	m.knownFilesLock.Unlock()
+
+	// Get the list of paths on disk
+	matches := m.finder.FindFiles()
+	m.consumeConcurrent(ctx, matches)
+	m.clearCurrentFingerprints()
```

Comment:
I think I asked you this before but I can't recall. Why can we do this in an asynchronous situation?